docs/content/docs/connectors/datastream/filesystem.md

```diff
@@ -797,3 +797,322 @@ before the job is restarted. This will result in your job not being able to rest
 pending part-files are no longer there and Flink will fail with an exception as it tries to fetch them and fails.
 
 {{< top >}}
+
+# File Source
+
```

Comment:
I would expect an introduction of the File Source, like we also have for File Sink. There is already quite some relevant content at https://nightlies.apache.org/flink/flink-docs-master/docs/dev/datastream/sources/, I would expect that some of that content to (also) appear here as introduction?

Reply:
Also look at the outline that was provided in https://issues.apache.org/jira/browse/FLINK-20188?focusedCommentId=17444601&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17444601

Reply:
@MartijnVisser Okay, I will make global adjustments and additions regarding the overall coherence of the article.
Regarding the File Source, I will reference the DataSource article to add the content.